phoenix-core/src/main/java/org/apache/phoenix/compile/UpsertCompiler.java

```diff
@@ -139,6 +140,12 @@ private static void setValues(byte[][] values, int[] pkSlotIndex, int[] columnIn
         for (int i = 0, j = numSplColumns; j < values.length; j++, i++) {
             byte[] value = values[j];
             PColumn column = table.getColumns().get(columnIndexes[i]);
+            if (value.length >= maxCellSizeBytes) {
```

Comment:
What about use-cases that use SINGLE_CELL_ARRAY_WITH_OFFSETS storage scheme wherein all the column keyvalues are stored inside a single cell? In that case, the hbase cell limit would apply to the sum of all the columns (or something along those lines). Either we should expand this to apply to all storage schemes, or restrict scope to only apply to ONE_CELL_PER_COLUMN